### ISSUE TITLE
[Bug]: Buffer pin not persist even with sessionoptions: globals

### ISSUE THREAD

**Issue by otavioschwanck** [akinsho/bufferline.nvim#677]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

The pins not persist betwee nsessions

### What did you expect to happen?

Buffer pins persist

### Config
```
return {
  {
    "akinsho/bufferline.nvim",
    event = "VeryLazy",
    keys = {
      { "<leader>bp", "<Cmd>BufferLineTogglePin<CR>",            desc = "Toggle pin" },
      { "<leader>bk", "<Cmd>BufferLineGroupClose ungrouped<CR>", desc = "Delete non-pinned buffers" },
      { "<leader>bn", ":e ~/.nvim-scratch<CR>",                  desc = "Open Scratch Buffer" },
      { "<C-s>",         "<cmd>BufferLineTogglePin<cr>",            desc = "Prev Buffer" },
      { "[b",         "<cmd>BufferLineCyclePrev<cr>",            desc = "Prev Buffer" },
      { "]b",         "<cmd>BufferLineCycleNext<cr>",            desc = "Next Buffer" },
      { "H",         "<cmd>BufferLineCyclePrev<cr>",            desc = "Prev Buffer" },
      { "L",         "<cmd>BufferLineCycleNext<cr>",            desc = "Next Buffer" },
      { "[B",         "<cmd>BufferLineMovePrev<cr>",             desc = "Switch Prev Buffer" },
      { "]B",         "<cmd>BufferLineMoveNext<cr>",             desc = "Switch Next Buffer" },
      { "<C-h>",         "<cmd>BufferLineMovePrev<cr>",             desc = "Switch Prev Buffer" },
      { "<C-l>",         "<cmd>BufferLineMoveNext<cr>",             desc = "Switch Next Buffer" },
    },
    opts = {
      options = {
        diagnostics = "nvim_lsp",
        always_show_bufferline = true,
        offsets = {
          {
            filetype = "neo-tree",
            text = "Neo-tree",
            highlight = "Directory",
            text_align = "left",
          },
        },
      },
    },
  }
}
```

### Additional Information

...
![session-bug](https://user-images.githubusercontent.com/9551316/219036508-517b291c-2be5-46c6-a2fa-c76074ce46dc.gif)
<img width="515" alt="image" src="https://user-images.githubusercontent.com/9551316/219036688-bb960480-5805-4803-9826-718b2373a169.png">

Session managed by:  https://github.com/folke/persistence.nvim

### commit

013857c12340bc14e285727eaf7d0bfb9807c768

**Comment by akinsho** [comment]
@otavioschwanck before I closed your previous issue I crosschecked locally with my config which uses autosession that pins are being reproduced when I reload the session. I currently don't have a lot of time to spend debugging issues so you will need to investigate further on your end what might be happening.

* Did you delete your old session before updating the sessionoptions? since if your old session had the old settings it wouldn't have worked.
* Did you check *inside* your session files to see if any globals are being saved?
* Dis you try and create a global yourself like `g:MyTestGlobal = "help"` and see if it is being reproduced when you restart the session.

These are all thing that will help narrow down the issue.
